lib/ansible/plugins/inventory/aws_ec2.py

```diff
@@ -315,6 +323,25 @@ def _boto3_conn(self, regions):
             Generator that yields a boto3 client and the region
         '''
 
+        if not regions:
+            try:
+                # as per https://boto3.amazonaws.com/v1/documentation/api/latest/guide/ec2-example-regions-avail-zones.html
+                client = boto3.client('ec2')
+                resp = client.describe_regions()
+                regions = resp.get('Regions', [])
+            except botocore.exceptions.NoRegionError:
```

Comment:
Do you know what versions were causing an issue? 
```
$ pip freeze | grep boto
boto==2.49.0
boto3==1.3.0
botocore==1.4.1
```
I tested back to 1.3.0 which was released in March, 2016. A lot of AWS modules need more up-to-date libraries, but if you can pinpoint what the issue was (or share a traceback so I can look for the problem) we could document that since using such an old version would omit so many of the regions and might cause problems for people if they assume all regions are being collected.

Reply:
i was told by 'people in irc' .. so i dont have specific versions